gap/generic/SnAnUnknownDegree.gi

```diff
@@ -0,0 +1,895 @@
+#############################################################################
+##
+##  This file is part of recog, a package for the GAP computer algebra system
+##  which provides a collection of methods for the constructive recognition
+##  of groups.
+##
+##  This files's authors include Friedrich Rober and Sergio Siccha.
+##
+##  Copyright of recog belongs to its developers whose names are too numerous
+##  to list here. Please refer to the COPYRIGHT file for details.
+##
+##  SPDX-License-Identifier: GPL-3.0-or-later
+##
+##
+##  This file provides code for recognising whether a permutation group
+##  is isomorphic to an alternating or symmetric group. It implements
+##  <Cite Key="JLNP13"/>.
+##
+#############################################################################
+#
+# FIXME: Move me into GAP
+# Helper function to compute all primes up to a given integer via a prime sieve
+RECOG.AllPrimesUpTo := function(n)
```

Comment:
I am actually not so sure that it is best to move this to GAP? It would then open the question whether GAP should cache these primes or not (which you want), and you'd be tied to that GAP version, and any fixes or improvements would be out of our hands.

I would suggest a slightly different interface anyway:

```gap
RECOG.PrimesCache := ShallowCopy(Primes); # all primes < 1000
RECOG.PrimesCacheUpperBound := 1000;
RECOG.CachePrimesUpTo := function(n)
    local p, ...
    if RECOG.PrimesCacheUpperBound >= n then return; fi;
    # Now run a sieve...
    ...
    RECOG.PrimesCacheUpperBound := n;
end;
```
Instead of storing `RECOG.PrimesCacheUpperBound`, one could also simply replace `n` by `2*n`, and then check `if Last(RECOG.PrimesCache) >= n then return; fi;`: by Bertrand's postulate, there is always a prime between n and 2*. T hen we compute more primes than asked for, but  ...
1. that's not too much of an issue for the numbers we are interested in,
2. you could compensate for this by implementing a common optimization to the sieve, namely skipping all even numbers from the start (halving the require memory for `sieve` again)
3. it actually can be useful if there is bad code which calls this function with N, N+1, N+2, ...: i.e., we kinda implemented "exponential backoff"
4. it can be improved by using refinements of Bertrand's postulate; e.g. it is also known that there is a prime between 3n and 4n for any natural number n; so we can do `n:=4*QuoInt(n+2,3);` which still gives us exponential backoff, just with a factor of 4/3 instead of 2.

Reply:
Ok, @FriedrichRober is rewriting that ATM. We would move this to `gap/tools.gi` and rename that file to `gap/utils.g`, or do you have a better idea?